stable_baselines3/common/env_checker.py

```diff
@@ -143,6 +143,10 @@ def _check_returned_values(env: gym.Env, observation_space: spaces.Space, action
 
     if isinstance(observation_space, spaces.Dict):
         assert isinstance(obs, dict), "The observation returned by `reset()` must be a dictionary"
+
+        if not obs.keys() == observation_space.spaces.keys():
+            raise AssertionError(f"The observation keys returned by `reset()` must match the observation space keys")
```

Comment:
you should also display the diffrence (current the f-string is not used)

Reply:
Something like 

    Observation space contains key 'foo', but not the observation returned by step()

?

Reply:
or simply: `f"{obs.keys()} != {observation_space.spaces.keys()}"`